### DISCUSSION TITLE
Expand selection to string keybinding

### DISCUSSION THREAD

**Discussion by olivroy** [posit-dev/positron#5003] [Q&A] [answered]
In RStudio, I use this a lot to expand selection.

![image](https://github.com/user-attachments/assets/4bbfb8c6-4b37-4da2-9496-7597a8398028)

Then I use Ctrl + Shift + Up to get this

![image](https://github.com/user-attachments/assets/191035e8-4f71-47f5-91ad-b364f36f300c)

and once again to get this

![image](https://github.com/user-attachments/assets/71437c07-daef-4cc4-9109-f75c9e267ddf)

Is there a similar command in Positron?

**Comment by jennybc** [comment]
In my hands, that exact key combination (Ctrl + Shift + Up) is having the desired effect in Positron as well. Have you definitely tried it?

**Reply by olivroy** [reply]
It does this.. I have selected : Map RStudio shortcuts.. I will investigate more. Lots of trials and errors.
![image](https://github.com/user-attachments/assets/ff8ccffc-290d-4ef4-9130-d689c1e79e44)

**Reply by jennybc** [reply]
Hmmm, that looks off by one (the position).

**Reply by jennybc** [reply]
But for the record, it's working for me on both macOS and Windows.

**Comment by DavisVaughan** [comment]
There's also a more powerful Expand Selection feature that we have support for that does this. Search for `editor.action.smartSelect.expand` in the Keybindings UI to see what it is mapped to for you.


https://github.com/user-attachments/assets/9bd50eba-596a-42cd-b695-1f26bf7f1d2f

**Comment by jennybc** [comment, marked as answer]
Sorry I misspoke. I was doing Ctrl + Shift + Right (not Up, oops). I.e. the experience I describe *is* what @DavisVaughan is talking about.

<img width="706" alt="Screenshot 2024-10-11 at 10 49 32 AM" src="https://github.com/user-attachments/assets/787986d4-cf2e-4ddf-8d6e-ee2d291557bc">

**Reply by olivroy** [reply]
Thanks! It works now!
